### DISCUSSION TITLE
Serverless Function Not Running, Gets Downloaded as Text File Instead

### DISCUSSION THREAD

**Discussion by gupta-ji6** [vercel/community#138] [Help] [answered]
## Problem Description 

I have written a serverless function to authorize users via Spotify.

### On Local
When I run it locally via the `vercel dev` command, it runs perfectly fine and redirects to a valid URL.

### On Prod

When I run the function on prod via the following link - https://usespotify.ayushgupta.tech/api/login - the function file gets downloaded weirdly. 🤨 
Also, the function logs on the dashboard don't have any activity when I open the link. I have set the required environment variables for the function to run.

I'm pretty sure I'm doing or missing something dumb (😬), kindly help me to fix this 🙏🏻 .

## Public Repo 

[gupta-ji6/useSpotify/api/login](https://github.com/gupta-ji6/useSpotify/blob/main/api/login.ts)

## Already Tried

1. Read [Why does my Serverless Function work locally but not when deployed?](https://vercel.com/support/articles/why-does-my-serverless-function-work-locally-but-not-when-deployed)
2. Prayed for it to work

**Comment by Pieparker** [comment, marked as answer]
Hey there. Is this still an issue for you? Adding a [`content-type`](https://vercel.com/docs/concepts/edge-network/headers#content-type) may help. Without it, perhaps the client isn't sure how to handle the response. That's my gut instinct but I'd be interested to see if it helps!
